pkg/cmd/create/workflow.go

```diff
@@ -0,0 +1,142 @@
+*/
+package create
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/gorhill/cronexpr"
+	"github.com/litmuschaos/litmusctl/pkg/apis"
+	"github.com/litmuschaos/litmusctl/pkg/utils"
+
+	"github.com/litmuschaos/litmus/litmus-portal/graphql-server/graph/model"
+
+	"github.com/spf13/cobra"
+)
+
+// workflowCmd represents the project command
+var workflowCmd = &cobra.Command{
+	Use: "workflow",
+	Short: `Create a Chaos Workflow
+	Example:
+	#create a chaos workflow
+	litmusctl create workflow -f workflow.yaml --project-id="d861b650-1549-4574-b2ba-ab754058dd04" --agent-id="1c9c5801-8789-4ac9-bf5f-32649b707a5c"
+
+	Note: The default location of the config file is $HOME/.litmusconfig, and can be overridden by a --config flag
+	`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		// Fetch user credentials
+		credentials, err := utils.GetCredentials(cmd)
+		utils.PrintError(err)
+
+		var chaosWorkFlowRequest model.ChaosWorkFlowRequest
+
+		workflowManifest, err := cmd.Flags().GetString("file")
+		utils.PrintError(err)
+
+		chaosWorkFlowRequest.ProjectID, err = cmd.Flags().GetString("project-id")
+		utils.PrintError(err)
+
+		// Handle blank input for project ID
+		if chaosWorkFlowRequest.ProjectID == "" {
+			utils.White_B.Print("\nEnter the Project ID: ")
+			fmt.Scanln(&chaosWorkFlowRequest.ProjectID)
+
+			if chaosWorkFlowRequest.ProjectID == "" {
+				utils.Red.Println("⛔ Project ID can't be empty!!")
+				os.Exit(1)
+			}
+		}
+
+		chaosWorkFlowRequest.ClusterID, err = cmd.Flags().GetString("agent-id")
+		utils.PrintError(err)
+
+		// Handle blank input for agent ID
+		if chaosWorkFlowRequest.ClusterID == "" {
+			utils.White_B.Print("\nEnter the Agent ID: ")
+			fmt.Scanln(&chaosWorkFlowRequest.ClusterID)
+
+			if chaosWorkFlowRequest.ClusterID == "" {
+				utils.Red.Println("⛔ Agent ID can't be empty!!")
+				os.Exit(1)
+			}
+		}
+
+		// Perform authorization
+		userDetails, err := apis.GetProjectDetails(credentials)
+		utils.PrintError(err)
+		var editAccess = false
+		var project apis.Project
+		for _, p := range userDetails.Data.Projects {
+			if p.ID == chaosWorkFlowRequest.ProjectID {
+				project = p
+			}
+		}
+		for _, member := range project.Members {
+			if (member.UserID == userDetails.Data.ID) && (member.Role == "Owner" || member.Role == "Editor") {
+				editAccess = true
+			}
+		}
+		if !editAccess {
+			utils.Red.Println("⛔ User doesn't have edit access to the project!!")
+			os.Exit(1)
+		}
+
+		// Parse workflow manifest and populate chaosWorkFlowInput
+		err = utils.ParseWorkflowManifest(workflowManifest, &chaosWorkFlowRequest)
+		if err != nil {
+			utils.Red.Println("❌ Error parsing workflow manifest: " + err.Error())
+			os.Exit(1)
+		}
+
+		// Make API call
+		createdWorkflow, err := apis.CreateWorkflow(chaosWorkFlowRequest, credentials)
+		if err != nil {
+			if (createdWorkflow.Data == apis.CreatedChaosWorkflow{}) {
+				if strings.Contains(err.Error(), "multiple write errors") {
```

Comment:
Why are we returning `multiple write errors` from GQL, are we not checking if workflow already exists in DB before scheduling it? @amityt @SarthakJain26 @Saranya-jena

Reply:
Resolving it for now as the change will be required in backend. Not a blocker for this PR.